Assets/Countly/Plugins/CountlySDK/Services/IViewModule.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+// interface for SDK users
+public interface IViewModule
+{
+    /// <summary>
+    /// Starts a view which would not close automatically
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <returns></returns>
+    string StartView(string viewName);
+
+    /// <summary>
+    /// Starts a view which would not close automatically
+    /// </summary>
+    /// <param name="viewName">name of the view</param>
+    /// <param name="viewSegmentation">segmentation that will be added to the view, set 'null' if none should be added</param>
+    /// <returns></returns>
+    string StartView(string viewName, Dictionary<string, object> viewSegmentation);
+
+    /// <summary>
+    /// Starts a view which would be closed automaticly
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <returns></returns>
+    string StartAutoStoppedView(string viewName);
+
+    /// <summary>
+    /// Starts a view which would be closed automaticly
+    /// </summary>
+    /// <param name="viewName"></param>
+    /// <param name="viewSegmentation"></param>
+    /// <returns></returns>
+    string StartAutoStoppedView(string viewName, Dictionary<string, object> viewSegmentation);
+
+    /// <summary>
+    /// Stops a view with the given name if it was open
+    /// </summary>
+    /// <param name="viewName">name of the view</param>
+    void StopViewWithName(string viewName);
```

Comment:
out of curiosity, why we are not using C# way of function overloading?
like
    void StopViewWithName(string viewName, Dictionary<string, object> viewSegmentation = null);

Reply:
Codacy hates it and I am not sure why 😂 but sure we can also do that, I still think it doesn't hurt and should be fine

Reply:
Yeah I think it will probably not hurt cuz we are doing checks